# Import Library

In [1]:
# Import Libraries
import os
import pandas as pd
import numpy as np
from scipy.stats import norm
from sklearn.preprocessing import StandardScaler
from scipy import stats
import warnings
warnings.filterwarnings('ignore')
%matplotlib inline

# Visuzliation Setting
import matplotlib.pyplot as plt
import matplotlib.font_manager as fm
from matplotlib import rc
from matplotlib import colors
import seaborn as sns

---

In [2]:
root = os.path.join(os.getcwd(), 'DATA')

# Training Data
빅콘테스트 제공 데이터 전처리

## Import Data

In [3]:
df_raw = pd.read_excel(os.path.join(root, 'train.xlsx'))
df_train = df_raw.copy()

### Add Columns

In [4]:
# 날짜변수 추가
df_train['year'] = df_train['REG_DATE'].dt.year
df_train['month'] = df_train['REG_DATE'].dt.month
df_train['day'] = df_train['REG_DATE'].dt.day

In [5]:
df_train['VALUE_COUNT'] = 0;
value_dict = {}
for name, value in zip(df_train['P_NAME'].value_counts().index,df_train['P_NAME'].value_counts()):
    value_dict[name] = value

def value(col):
    return value_dict[col]

df_train['VALUE_COUNT'] = df_train['P_NAME'].apply(value)

### One hot encoding
P_IMPORT_TYPE 이라는 특수한 컬럼에 대한 전처리

In [6]:
import_type_list = set()
for tmp in df_train.P_IMPORT_TYPE.unique():
    for a in tmp.split(','):
        import_type_list.add(a)

In [7]:
for name in import_type_list:
    df_train[name] = 0
    df_train.loc[df_train['P_IMPORT_TYPE'].str.contains(name, regex=False), name] = 1

---

# Weather data

## Import Data

In [8]:
df_weather_code = pd.read_excel(os.path.join(root, 'raw_weather_code.xlsx'), header=None, index_col=0)

In [9]:
weather_list = [pd.read_csv(os.path.join(root, 'raw_weather_20151228_20161227.csv'), encoding='euc-kr') , 
                pd.read_csv(os.path.join(root, 'raw_weather_20161228_20171227.csv'), encoding='euc-kr') , 
                pd.read_csv(os.path.join(root, 'raw_weather_20171228_20181227.csv'), encoding='euc-kr') , 
                pd.read_csv(os.path.join(root, 'raw_weather_20181228_20191227.csv'), encoding='euc-kr') , 
                pd.read_csv(os.path.join(root, 'raw_weather_20191228_20201227.csv'), encoding='euc-kr') , 
                pd.read_csv(os.path.join(root, 'raw_weather_20201228_20210818.csv'), encoding='euc-kr')]

## Preprocess
- '지점'에 따른 나라명 컬럼 (country) 추가
- 각 나라, 일자 별로 평균 강수량, 풍속, 기온 계산

In [10]:
# 지점에 따라 나라명 추가
def set_country(row):
    data = df_weather_code[df_weather_code[1] == row['지점']]
    if data.empty:
        return ""
    return data.iloc[0][2]


def preprocess_weather(df_weather):
    # 날짜 정보 정리
    df_weather['year'] = df_weather['일시'].astype('str').str[:4].astype('int')
    df_weather['month'] = df_weather['일시'].astype('str').str[5:7].astype('int')
    df_weather['day'] = df_weather['일시'].astype('str').str[8:10].astype('int')
    # 1차 평균
    df_weather['rain'] = df_weather[['지점', 'year', 'month', 'day', '강수량']].groupby(['지점', 'year', 'month', 'day']).transform('mean')
    df_weather['wind'] = df_weather[['지점', 'year', 'month', 'day', '풍속']].groupby(['지점', 'year', 'month', 'day']).transform('mean')
    df_weather['temperature'] = df_weather[['지점', 'year', 'month', 'day', '기온']].groupby(['지점', 'year', 'month', 'day']).transform('mean')
    # 컬럼/행 정리
    df_weather.drop(columns = ['지점명', '일시', '강수량', '풍속', '기온'], inplace=True)
    df_weather.drop_duplicates(inplace=True)
    # 나라명 추가
    df_weather['CTRY_1'] = ""
    for i, row in df_weather.iterrows():
        df_weather.at[i, 'CTRY_1'] = set_country(row)
    # 2차 평균
    df_weather['rain'] = df_weather[['CTRY_1', 'year', 'month', 'day', 'rain']].groupby(['CTRY_1', 'year', 'month', 'day']).transform('mean')
    df_weather['wind'] = df_weather[['CTRY_1', 'year', 'month', 'day', 'wind']].groupby(['CTRY_1', 'year', 'month', 'day']).transform('mean')
    df_weather['temperature'] = df_weather[['CTRY_1', 'year', 'month', 'day', 'temperature']].groupby(['CTRY_1', 'year', 'month', 'day']).transform('mean')
    # 컬럼/행 정리
    df_weather.drop(columns = ['지점'], inplace=True)
    df_weather.drop_duplicates(inplace=True)
    # 인덱스 정리
    df_weather.reset_index(drop=True, inplace=True)

In [11]:
for df in weather_list:
    preprocess_weather(df)
df_weather = pd.concat(weather_list)

In [12]:
df_weather = pd.concat(weather_list)

In [13]:
df_weather.describe()

year         month           day          rain          wind  \
count  13947.000000  13947.000000  13947.000000  10303.000000  13947.000000   
mean    2018.258407      6.276762     15.697426    -20.478450      3.312765   
std        1.613987      3.414373      8.809565     78.401760      5.434442   
min     2015.000000      1.000000      1.000000   -999.000000   -156.116667   
25%     2017.000000      3.000000      8.000000     -6.701488      2.031250   
50%     2018.000000      6.000000     16.000000      1.035000      2.625000   
75%     2020.000000      9.000000     23.000000      3.676250      5.434846   
max     2021.000000     12.000000     31.000000    915.000000     15.125000   

        temperature  
count  13947.000000  
mean      16.499339  
std       10.954919  
min      -26.175000  
25%        9.912083  
50%       19.356250  
75%       25.244375  
max       33.275000

#### 확인필요
- outlier 찾아내기 > 값이 너무 크거나 작은 경우 제외
- 날짜별로 확인 후 비어있는 값 채워넣기 (전/다음날 이용)
    - 13947개 데이터 중 rain, wind, temperature 갯수 보면 몇개 비어있는지 확인 가능
- 합치기..
- 강수량의 경우 NaN 값이 너무 많음
    - 위 전처리들을 한 후에도 많다면 사용불가
    - 위 전처리 후에는 적다면 전날/다음날 데이터 기반으로 채워넣기 진행

## Add to Training Data

In [14]:
df_train = pd.merge(df_train, df_weather, how='left', on=['year', 'month', 'day', 'CTRY_1'])

In [15]:
pd.set_option("display.max_columns", None)
df_train.sample(5)

REG_DATE P_TYPE CTRY_1 CTRY_2 P_PURPOSE CATEGORY_1   CATEGORY_2  \
35262 2019-05-13    수산물  아르헨티나     중국       판매용   연체류 해물모듬          오징어   
34351 2019-04-08    수산물    러시아     일본       판매용      알 곤이류          성게알   
35738 2019-05-27    수산물  시에라리온  시에라리온       판매용         어류    서대 박대 페루다   
32855 2019-02-18    수산물    세네갈    세네갈       판매용   연체류 해물모듬         갑오징어   
31396 2018-12-24    수산물     중국     중국       판매용         어류  조기 보구치 강다리    

      P_NAME P_IMPORT_TYPE     P_PRICE  year  month  day  VALUE_COUNT  창난  줄기  \
35262    오징어            냉동    4.511682  2019      5   13         2150   0   0   
34351    성게알         냉장,횟감  241.875480  2019      4    8          453   0   0   
35738     서대            냉동    3.984615  2019      5   27          359   0   0   
32855   갑오징어            냉동    4.600000  2019      2   18          783   0   0   
31396    참조기            냉동    9.478136  2018     12   24          206   0   0   

       외투막  필렛(F)  지느러미  절단  한쪽껍질붙은  볼살  캐비아대용  꼬리_외화획득용  포장횟감  내장  횟감  머리  활  \
35262    0      0     0   0       0   0      0         0     0   0   0   0  0   
34351    0      0     0   0       0   0      0         0     0   0   1   0  0   
35738    0      0     0   0       0   0      0         0     0   0   0   0  0   
32855    0      0     0   0       0   0      0         0     0   0   0   0  0   
31396    0      0     0   0       0   0      0         0     0   0   0   0  0   

       알  머리살  난포선  건조  껍질  개아지살  집게다리  간  다리  냉동  눈살  냉장  염장  살  슬라이스(S)  동체  \
35262  0    0    0   0   0     0     0  0   0   1   0   0   0  0        0   0   
34351  0    0    0   0   0     0     0  0   0   0   0   1   0  0        0   0   
35738  0    0    0   0   0     0     0  0   0   1   0   0   0  0        0   0   
32855  0    0    0   0   0     0     0  0   0   1   0   0   0  0        0   0   
31396  0    0    0   0   0     0     0  0   0   1   0   0   0  0        0   0   

       곤이  훈제  머리_외화획득용  자숙  목살  턱살      rain      wind  temperature  
35262   0   0         0   0   0   0       NaN       NaN          NaN  
34351   0   0         0   0   0   0       NaN       NaN          NaN  
35738   0   0         0   0   0   0       NaN       NaN          NaN  
32855   0   0         0   0   0   0       NaN       NaN          NaN  
31396   0   0         0   0   0   0  1.329386  2.255651    -0.703571

---

# Salinity

## Import Data

In [16]:
# df_salinity = pd.read_csv(os.path.join(root, 'raw_salinity.csv'))

## Drop Columns & Rows

In [17]:
# df_salinity.drop(df_salinity.columns[2], inplace=True, axis=1)
# df_salinity = df_salinity[(2015 <= df_salinity['obs_year']) & (df_salinity['obs_year'] <= 2021)]

2020, 2021 데이터의 부재로 인해 보류

---

# Oil

## Import Data

In [18]:
df_oil = pd.read_csv(os.path.join(root, 'raw_oil.csv'))
df_oil_dubai = pd.read_csv(os.path.join(root, 'raw_oil_dubai.csv'))
df_oil_brent = pd.read_csv(os.path.join(root, 'raw_oil_brent.csv'))

## Preprocess

In [19]:
def preprocess_oil(df):
    df['year'] = df['날짜'].str[:4].astype('int')
    df['month'] = df['날짜'].str[6:8].astype('int')
    df['day'] = df['날짜'].str[10:12].astype('int')
#     df['date'] = pd.to_datetime(df['날짜'].str[:4] + df['날짜'].str[6:8] + df['날짜'].str[10:12])
#     df = df.query('date.dt.dayofweek == 0')
    df.drop(columns = ['날짜', '오픈', '고가', '저가', '거래량', '변동 %'], inplace=True, axis=1)
    df.sort_index(ascending=False)
    return df

In [20]:
df_oil = preprocess_oil(df_oil)
df_oil_dubai = preprocess_oil(df_oil_dubai)
df_oil_brent = preprocess_oil(df_oil_brent)

## Check

In [21]:
print("oil:", len(df_oil), "\ndubai:", len(df_oil_dubai), "\nbrent:", len(df_oil_brent))

oil: 1451 
dubai: 1383 
brent: 1420


oil과 비교하였을 때 dubai의 경우 28개, brent의 경우 3개의 데이터가 적다  
따라서 df_oil 데이터를 사용하도록 한다

## Add to Training Data

In [22]:
df_train = pd.merge(df_train, df_oil, how='left', on=['year', 'month', 'day'])

In [23]:
pd.set_option("display.max_columns", None)
df_train.sample(5)

REG_DATE P_TYPE CTRY_1 CTRY_2 P_PURPOSE   CATEGORY_1 CATEGORY_2  \
3800  2016-05-09    수산물     중국     중국       판매용           어류     메기 동자개   
29453 2018-10-22    수산물  인도네시아  인도네시아       판매용  젓갈류 해조류 해파리        해파리   
28819 2018-10-01    수산물   불가리아   불가리아       판매용       패류 멍게류         고동   
19793 2017-11-27    수산물     미국     미국       판매용           어류        가오리   
30150 2018-11-12    수산물     태국     태국       판매용     연체류 해물모듬       갑오징어   

      P_NAME P_IMPORT_TYPE   P_PRICE  year  month  day  VALUE_COUNT  창난  줄기  \
3800     동자개             활  5.926949  2016      5    9          206   0   0   
29453    해파리            염장  1.905328  2018     10   22          265   0   0   
28819   피뿔고둥       냉동,살,자숙  8.559476  2018     10    1          246   0   0   
19793    가오리            냉동  2.567153  2017     11   27          803   0   0   
30150   갑오징어            냉장  9.744136  2018     11   12          783   0   0   

       외투막  필렛(F)  지느러미  절단  한쪽껍질붙은  볼살  캐비아대용  꼬리_외화획득용  포장횟감  내장  횟감  머리  활  \
3800     0      0     0   0       0   0      0         0     0   0   0   0  1   
29453    0      0     0   0       0   0      0         0     0   0   0   0  0   
28819    0      0     0   0       0   0      0         0     0   0   0   0  0   
19793    0      0     0   0       0   0      0         0     0   0   0   0  0   
30150    0      0     0   0       0   0      0         0     0   0   0   0  0   

       알  머리살  난포선  건조  껍질  개아지살  집게다리  간  다리  냉동  눈살  냉장  염장  살  슬라이스(S)  동체  \
3800   0    0    0   0   0     0     0  0   0   0   0   0   0  0        0   0   
29453  0    0    0   0   0     0     0  0   0   0   0   0   1  0        0   0   
28819  0    0    0   0   0     0     0  0   0   1   0   0   0  1        0   0   
19793  0    0    0   0   0     0     0  0   0   1   0   0   0  0        0   0   
30150  0    0    0   0   0     0     0  0   0   0   0   1   0  0        0   0   

       곤이  훈제  머리_외화획득용  자숙  목살  턱살      rain      wind  temperature     종가  
3800    0   0         0   0   0   0  2.527439  2.363924    17.880967  43.44  
29453   0   0         0   0   0   0       NaN       NaN          NaN  69.17  
28819   0   0         0   1   0   0       NaN       NaN          NaN  75.30  
19793   0   0         0   0   0   0       NaN       NaN          NaN  58.11  
30150   0   0         0   0   0   0  6.377778  1.391667    28.003333  59.93

---

# Korea Weather

## Import Data

In [24]:
df_weather_kr = pd.read_csv(os.path.join(root, 'raw_weather_korea.csv'), encoding='euc-kr')

## Group
일자 별로 평균 구하기

In [25]:
df_weather_kr['wind_kr'] = df_weather_kr[['일시', '평균 풍속(m/s)']].groupby(['일시']).transform('mean')
df_weather_kr['temperature_kr'] = df_weather_kr[['일시', '평균 기온(°C)']].groupby(['일시']).transform('mean')
df_weather_kr['water_temp_kr'] = df_weather_kr[['일시', '평균 수온(°C)']].groupby(['일시']).transform('mean')

## Preprocess Date

In [26]:
df_weather_kr['year'] = df_weather_kr['일시'].str[:4].astype('int')
df_weather_kr['month'] = df_weather_kr['일시'].str[5:7].astype('int')
df_weather_kr['day'] = df_weather_kr['일시'].str[8:].astype('int')

## Drop Column
어차피 전부 해안가 대한민국이라 지점은 필요없다  
사용한 컬럼은 제거한다

In [27]:
drop = ['지점', '일시', '평균 풍속(m/s)', '평균 기온(°C)', '평균 수온(°C)']

In [28]:
df_weather_kr.drop(columns=drop, inplace=True, axis=1)

## Add to Training Data

In [29]:
df_train = pd.merge(df_train, df_weather_kr, how='left', on=['year', 'month', 'day'])

In [30]:
pd.set_option("display.max_columns", None)
df_train.sample(5)

REG_DATE P_TYPE CTRY_1 CTRY_2 P_PURPOSE CATEGORY_1 CATEGORY_2 P_NAME  \
88122  2016-07-04    수산물     중국     중국       판매용         어류         복어    흑밀복   
659255 2019-10-07    수산물    베트남    베트남       판매용   연체류 해물모듬       갑오징어   갑오징어   
339191 2017-12-18    수산물     중국     중국       판매용         어류         복어     참복   
553961 2019-03-04    수산물  인도네시아  인도네시아       판매용   연체류 해물모듬         문어     문어   
630340 2019-08-05    수산물   이탈리아     일본       판매용         어류    참치 새치류    참다랑어   

       P_IMPORT_TYPE    P_PRICE  year  month  day  VALUE_COUNT  창난  줄기  외투막  \
88122             냉동   0.918242  2016      7    4          131   0   0    0   
659255            냉동   4.020221  2019     10    7          783   0   0    0   
339191            냉동   3.521023  2017     12   18           15   0   0    0   
553961            냉동   6.473073  2019      3    4          486   0   0    0   
630340   냉동,필렛(F),횟감  33.155360  2019      8    5          979   0   0    0   

        필렛(F)  지느러미  절단  한쪽껍질붙은  볼살  캐비아대용  꼬리_외화획득용  포장횟감  내장  횟감  머리  활  알  \
88122       0     0   0       0   0      0         0     0   0   0   0  0  0   
659255      0     0   0       0   0      0         0     0   0   0   0  0  0   
339191      0     0   0       0   0      0         0     0   0   0   0  0  0   
553961      0     0   0       0   0      0         0     0   0   0   0  0  0   
630340      1     0   0       0   0      0         0     0   0   1   0  0  0   

        머리살  난포선  건조  껍질  개아지살  집게다리  간  다리  냉동  눈살  냉장  염장  살  슬라이스(S)  동체  \
88122     0    0   0   0     0     0  0   0   1   0   0   0  0        0   0   
659255    0    0   0   0     0     0  0   0   1   0   0   0  0        0   0   
339191    0    0   0   0     0     0  0   0   1   0   0   0  0        0   0   
553961    0    0   0   0     0     0  0   0   1   0   0   0  0        0   0   
630340    0    0   0   0     0     0  0   0   1   0   0   0  0        0   0   

        곤이  훈제  머리_외화획득용  자숙  목살  턱살       rain      wind  temperature     종가  \
88122    0   0         0   0   0   0   4.915931  2.625000    24.874548  48.65   
659255   0   0         0   0   0   0 -85.827083  2.250000    29.042188  52.75   
339191   0   0         0   0   0   0  -9.881667  2.358725    -1.692473  57.16   
553961   0   0         0   0   0   0        NaN       NaN          NaN  56.59   
630340   0   0         0   0   0   0        NaN       NaN          NaN  54.69   

         wind_kr  temperature_kr  water_temp_kr  
88122   5.035294       21.864706      20.788235  
659255  5.250000       19.792857      22.525000  
339191  6.994118        6.105882      13.017647  
553961  5.517647        9.523529      10.911765  
630340  3.252941       26.406250      26.218750

---

# 소비자물가지수

## Import Data

In [31]:
df_cpi_tmp1 = pd.read_excel(os.path.join(root, 'raw_cpi_region.xlsx'), encoding='euc-kr')  # 시도별 CPI
df_cpi_tmp2 = pd.read_excel(os.path.join(root, 'raw_cpi_purpose.xlsx'), encoding='euc-kr')  # 품목별 CPI

## Drop Columns/Rows

In [32]:
df_cpi_tmp1 = df_cpi_tmp1[df_cpi_tmp1['시도별(1)'] == '전국'].drop(columns=['시도별(1)', '시도별(2)','품목별(1)'], axis=1)

In [33]:
df_cpi_tmp2 = df_cpi_tmp2.iloc[[2,4],:].drop(columns=['시도별'],axis=1)  # 식료품 및 음식서비스 컬럼 추출 &시도별 칼럼 제거

## Pivoting

In [34]:
df_cpi1 = pd.DataFrame(df_cpi_tmp1.values.reshape(49,1),index=df_cpi_tmp1.columns, columns=['CPI'])

In [35]:
df_cpi2 = pd.DataFrame(df_cpi_tmp2.iloc[0,:].values,index = df_cpi_tmp2.columns,columns=['CPI_food'])
df_cpi2['CPI_foodservice'] = df_cpi_tmp2.iloc[1,:].values

## Preprocess Date

In [36]:
df_cpi2.drop('지출목적별',axis=0,inplace=True) # 지출목적별 행 삭제

In [37]:
df_cpi1.reset_index(inplace=True)  # 날짜 인덱스 > 컬럼으로 변경
df_cpi2.reset_index(inplace=True)  # 날짜 인덱스 > 컬럼으로 변경

In [38]:
df_cpi1['year'] = df_cpi1['index'].str[:4].astype('int')
df_cpi1['month'] = df_cpi1['index'].str[5:].astype('int')
df_cpi1.drop(columns=['index'], inplace=True, axis=1)

df_cpi2['year'] = df_cpi2['index'].str[:4].astype('int')
df_cpi2['month'] = df_cpi2['index'].str[5:].astype('int')
df_cpi2.drop(columns=['index'], inplace=True, axis=1)

## Add to Training Data

In [39]:
df_train = pd.merge(df_train, df_cpi1, how='left', on=['year', 'month'])
df_train = pd.merge(df_train, df_cpi2, how='left', on=['year', 'month'])

In [40]:
pd.set_option("display.max_columns", None)
df_train.sample(5)

REG_DATE P_TYPE CTRY_1 CTRY_2 P_PURPOSE CATEGORY_1  CATEGORY_2  \
701223 2019-12-23    수산물    러시아    러시아       판매용        갑각류           게   
541064 2019-01-28    수산물  아르헨티나  아르헨티나       판매용         어류          홍어   
193647 2017-02-20    수산물   뉴질랜드   뉴질랜드       판매용      알 곤이류  호키(새꼬리민태)알   
234541 2017-05-15    수산물    러시아    러시아       판매용     패류 멍게류    조개 백합 대합   
65040  2016-05-16    수산물    베트남    베트남       판매용   연체류 해물모듬         오징어   

        P_NAME P_IMPORT_TYPE    P_PRICE  year  month  day  VALUE_COUNT  창난  \
701223      대게            냉장   6.248869  2019     12   23          603   0   
541064      홍어            냉동   3.703274  2019      1   28          746   0   
193647  새꼬리민태알          냉동,알   2.878231  2017      2   20           74   0   
234541    북방대합             활   2.650000  2017      5   15          134   0   
65040    화살오징어            건조  25.773991  2016      5   16          512   0   

        줄기  외투막  필렛(F)  지느러미  절단  한쪽껍질붙은  볼살  캐비아대용  꼬리_외화획득용  포장횟감  내장  횟감  \
701223   0    0      0     0   0       0   0      0         0     0   0   0   
541064   0    0      0     0   0       0   0      0         0     0   0   0   
193647   0    0      0     0   0       0   0      0         0     0   0   0   
234541   0    0      0     0   0       0   0      0         0     0   0   0   
65040    0    0      0     0   0       0   0      0         0     0   0   0   

        머리  활  알  머리살  난포선  건조  껍질  개아지살  집게다리  간  다리  냉동  눈살  냉장  염장  살  \
701223   0  0  0    0    0   0   0     0     0  0   0   0   0   1   0  0   
541064   0  0  0    0    0   0   0     0     0  0   0   1   0   0   0  0   
193647   0  0  1    0    0   0   0     0     0  0   0   1   0   0   0  0   
234541   0  1  0    0    0   0   0     0     0  0   0   0   0   0   0  0   
65040    0  0  0    0    0   1   0     0     0  0   0   0   0   0   0  0   

        슬라이스(S)  동체  곤이  훈제  머리_외화획득용  자숙  목살  턱살  rain    wind  temperature  \
701223        0   0   0   0         0   0   0   0   NaN     NaN          NaN   
541064        0   0   0   0         0   0   0   0   NaN     NaN          NaN   
193647        0   0   0   0         0   0   0   0   NaN     NaN          NaN   
234541        0   0   0   0         0   0   0   0   NaN     NaN          NaN   
65040         0   0   0   0         0   0   0   0   9.8  2.1875     27.46875   

           종가    wind_kr  temperature_kr  water_temp_kr     CPI CPI_food  \
701223  60.52   7.052381        9.338095      13.947619  105.12   109.56   
541064  51.99   8.952941        6.525000      11.262500  104.24   109.22   
193647  53.97  11.593750        6.112500      10.173333  102.92   107.73   
234541  48.85   5.075000       15.850000      15.362500  102.83   105.34   
65040   47.72   6.647059       14.923529      15.100000  100.81   101.39   

       CPI_foodservice  
701223          110.52  
541064          109.86  
193647          104.11  
234541          104.88  
65040           102.43

---

# Final

## One-hot Encoding

In [41]:
one_hot = ['CTRY_1', 'CTRY_2', 'P_PURPOSE', 'CATEGORY_1', 'CATEGORY_2', 'P_NAME']

In [42]:
df_train = pd.get_dummies(df_train, columns=one_hot)

## Drop Columns

In [43]:
drop = ['REG_DATE', 'P_TYPE', 'P_IMPORT_TYPE']

In [44]:
df_train.drop(columns = drop, inplace=True)

In [45]:
pd.set_option("display.max_columns", None)
df_train.sample(5)

P_PRICE  year  month  day  VALUE_COUNT  창난  줄기  외투막  필렛(F)  지느러미  절단  \
128405  1.022522  2016     10    3         1640   0   0    0      0     0   0   
509080  2.752600  2018     11   26          131   0   0    0      0     0   0   
694381  1.433445  2019     12   16          143   0   0    0      0     0   0   
175456  9.397731  2017      1    6          783   0   0    0      0     0   0   
346176  0.920687  2018      1    1           27   0   0    0      0     0   0   

        한쪽껍질붙은  볼살  캐비아대용  꼬리_외화획득용  포장횟감  내장  횟감  머리  활  알  머리살  난포선  건조  껍질  \
128405       0   0      0         0     0   0   0   0  0  0    0    0   0   0   
509080       0   0      0         0     0   0   0   0  0  0    0    0   0   0   
694381       0   0      0         0     0   0   0   0  0  0    0    0   0   0   
175456       0   0      0         0     0   0   0   0  0  0    0    0   0   0   
346176       0   0      0         0     0   0   0   0  0  0    0    0   0   0   

        개아지살  집게다리  간  다리  냉동  눈살  냉장  염장  살  슬라이스(S)  동체  곤이  훈제  머리_외화획득용  \
128405     0     0  0   0   1   0   0   0  0        0   0   0   0         0   
509080     0     0  0   0   1   0   0   0  0        0   0   0   0         0   
694381     0     0  0   0   1   0   0   0  0        0   0   0   0         0   
175456     0     0  0   0   1   0   0   0  0        0   1   0   0         0   
346176     0     0  0   0   1   0   0   0  0        0   0   0   0         0   

        자숙  목살  턱살      rain     wind  temperature     종가   wind_kr  \
128405   0   0   0       NaN      NaN          NaN  48.81  4.547059   
509080   0   0   0  1.238889  2.47943     5.562342  51.63  3.635294   
694381   0   0   0       NaN      NaN          NaN  60.21  5.628571   
175456   0   0   0  1.411111  1.84375    24.520312  53.99  6.076471   
346176   0   0   0       NaN      NaN          NaN  60.24  6.629412   

        temperature_kr  water_temp_kr     CPI CPI_food CPI_foodservice  \
128405       23.335294      23.123529  101.60   104.76          103.06   
509080       13.343750      16.435294  104.71   109.19          109.15   
694381       13.000000      14.752381  105.12   109.56          110.52   
175456        9.611765      13.156250  102.64    107.6          103.78   
346176        5.247059      11.882353  103.42   106.42          106.58   

        CTRY_1_가나  CTRY_1_감비아  CTRY_1_그리스  CTRY_1_그린란드  CTRY_1_기니  \
128405          0           0           0            0          0   
509080          0           0           0            0          0   
694381          0           0           0            0          0   
175456          0           0           0            0          0   
346176          0           0           0            0          0   

        CTRY_1_기니비사우  CTRY_1_나미비아  CTRY_1_남아프리카 공화국  CTRY_1_네덜란드  CTRY_1_노르웨이  \
128405             0            0                 0            0            0   
509080             0            0                 0            0            0   
694381             0            0                 0            0            0   
175456             0            0                 0            0            0   
346176             0            0                 0            0            0   

        CTRY_1_뉴질랜드  CTRY_1_니카라과  CTRY_1_대만  CTRY_1_대한민국  CTRY_1_라이베리아  \
128405            0            0          0            0             0   
509080            0            0          0            0             0   
694381            0            0          0            0             0   
175456            0            0          0            0             0   
346176            1            0          0            0             0   

        CTRY_1_라트비아  CTRY_1_러시아  CTRY_1_루마니아  CTRY_1_리비아  CTRY_1_마다가스카르  \
128405            0           1            0           0              0   
509080            0           0            0           0              0   
694381            0           0            0           0              0   
175456            0           0          

## Save Files

In [46]:
def save_file(df, file_name):
    df.to_csv(os.path.join(root, file_name), encoding='utf-8')

In [48]:
save_file(df_weather, 'preprocessed_weather.csv')  # 제조국 날씨
save_file(df_oil, 'preprocessed_oil.csv')  # 원유 종가
save_file(df_weather_kr, 'preprocessed_weather_korea.csv')  # 한국 날씨
save_file(df_cpi1, 'preprocessed_cpi_region.csv')  # 전체 소비자물가지수
save_file(df_cpi2, 'preprocessed_cpi_purpose.csv')  # 음식/음식서비스 소비자물가지수
save_file(df_train, 'preprocessed_train.csv')  # 최종 df